PandorasBox/Features/UI/WorkshopHelper.cs

```diff
@@ -0,0 +1,612 @@
+                if (ImGui.Checkbox($"{i + 1}", ref configValue)) { Workshops[i] = configValue; }
+                if (i != Workshops.Count - 1) ImGui.SameLine();
+            }
+            ImGui.SameLine();
+            if (ImGui.Button("Deselect"))
+                Workshops.Keys.ToList().ForEach(key => Workshops[key] = false);
+
+            try
+            {
+                if (IsInsufficientRank)
+                {
+                    ImGui.BeginDisabled();
+                    ImGui.TextColored(ImGuiColors.DalamudRed, "Insufficient Rank (items marked with *)");
+                }
+                if (ImGui.Button("Execute Schedule"))
+                {
+                    CurrentWorkshop = Workshops.FirstOrDefault(pair => pair.Value).Key;
+                    var restDays = GetCurrentRestDays();
+                    if (restDays.Contains(SelectedCycle - 1))
+                        PrintPluginMessage("Selected cycle is a rest day. Cannot schedule on a rest day.");
+                    else
+                        ScheduleList();
+                }
+                if (IsInsufficientRank)
+                    ImGui.EndDisabled();
+            }
+            catch (Exception e) { PluginLog.Log(e.ToString()); return; }
+        }
+
+        public static unsafe Vector2 GetNodePosition(AtkResNode* node)
+        {
+            var pos = new Vector2(node->X, node->Y);
+            var par = node->ParentNode;
+            while (par != null)
+            {
+                pos *= new Vector2(par->ScaleX, par->ScaleY);
+                pos += new Vector2(par->X, par->Y);
+                par = par->ParentNode;
+            }
+
+            return pos;
+        }
+
+        public static unsafe Vector2 GetNodeScale(AtkResNode* node)
+        {
+            if (node == null) return new Vector2(1, 1);
+            var scale = new Vector2(node->ScaleX, node->ScaleY);
+            while (node->ParentNode != null)
+            {
+                node = node->ParentNode;
+                scale *= new Vector2(node->ScaleX, node->ScaleY);
+            }
+
+            return scale;
+        }
+
+        internal static List<Item> ScheduleImport(List<string> rawItemStrings)
+        {
+            List<Item> items = ParseItems(rawItemStrings);
+            CopiedSchedule = items;
+            return CopiedSchedule;
+        }
+
+        public static List<Item> ParseItems(List<string> itemStrings)
+        {
+            List<Item> items = new List<Item>();
+            foreach (var itemString in itemStrings)
+            {
+                bool matchFound = false;
+                foreach (var craftable in Craftables)
+                {
+                    string craftableNoPrefix = craftable.Name.Replace("Isleworks ", "")
+                        .Replace("Isleberry ", "")
+                        .Replace("Islefish ", "");
+                    if (itemString.ToLower().Contains(craftableNoPrefix.ToLower()))
```

Comment:
Trying to import an Isleworks Garnet Rapier will also add an Islefish Pie, presumably because "Rapier" contains "pie" as a substring. There may be more cases where this could happen, but I haven't checked.

Reply:
fixed in 2de3a9, hopefully via
```c#
private static bool IsMatch(string x, string y)
{
    string pattern = $@"\b{Regex.Escape(y)}\b";
    return Regex.IsMatch(x, pattern);
}
```
Had to limit the matching agnosticism I had in mind by matching against whole words but this fixed the pie -> rapier issue. Let me know if you encounter any more.